I18n

pascalaubry opened the pull request:
Made all the application support EN/FR and added the necessary to add future languages (6 experimental locales translated by an AI).

pascalaubry committed:
Upgraded Jinja2 to 3.1.5

pascalaubry committed:
Added SVG flags

pascalaubry committed:
Removed all the references to requirements.txt.

pascalaubry committed:
Added the i18n_update.py script to update the translation files.

pascalaubry committed:
Deleted 96-i18n.md

pascalaubry committed:
Added the basic routines for i18n

pascalaubry committed:
Added translation files and the babel configuration

pascalaubry committed:
Removed unused utils/i18n.py

pascalaubry committed:
Translated all the strings in Python code

pascalaubry committed:
Translated all the strings in HTML files

pascalaubry committed:
Version 2.4.19
Fixed PapiWebConfig.copyright usage

pascalaubry committed:
Set the locale before launching the web browser

pascalaubry committed:
Added missing translations

pascalaubry committed:
Improved the i18n update tool

pascalaubry committed:
Added a FAQ for the web interface.

pascalaubry committed:
Simplified i18n.py

pascalaubry committed:
Simplified i18n.py

pascalaubry committed:
Added missing translations

pascalaubry committed:
Fixed color on timer cards

pascalaubry committed:
Fixed cancel button text

pascalaubry committed:
Added missing parameters

pascalaubry committed:
Fixed timers tab text

pascalaubry committed:
Fixed fr translation

pascalaubry committed:
Improved fr translation

pascalaubry committed:
Improved fr translation

pascalaubry committed:
Added MO files to the EXE

pascalaubry committed:
Removed unneeded translations

pascalaubry committed:
Removed unneeded translations

pascalaubry committed:
Fixed translation details in controllers
Reorganized the /utils folder
Added more translations
Added automatic translation for new locales
Added option [i18n].experimental_locales
Improved the translation reports of the documentation

pascalaubry committed:
Fixed issue in export.py

pascalaubry committed:
Removed locales fi, is and pt (too poor automatic translations)
Improved the locale dropdown menus

pascalaubry committed:
Fixed fr locale

pascalaubry committed:
Improved the selection of the number of columns with a dropdown menu

pascalaubry committed:
Improved the locale dropdown menus
Update README.md from export.py

pascalaubry committed:
Documented option i18n.experimental_locales

pascalaubry committed:
Improved the layout of screen and family modals

pascalaubry committed:
Translated batch files in English

pascalaubry committed:
Updated the ChangeLog

pascalaubry committed:
Moved generate_papi_template to /utils/papi

pascalaubry committed:
Added a warning icon with the flags of experimental locales

pascalaubry committed:
Use an ordered dict for locales to keep the trusted locales at the beginning of translation reports

pascalaubry committed:
Added babel dependency

pascalaubry committed:
Added Unidecode dependency

timothyarmes committed:
Correct/improve English text

pascalaubry committed:
Fixed links in 86-i18n.md

pascalaubry committed:
Added dependencies for developers

pascalaubry committed:
Updated the roadmap with multilingual support

pascalaubry committed:
Updated the versions of all the dependencies

pascalaubry committed:
Added /src to the path in i18n_update.py

pascalaubry committed:
Added missing FR translation

pascalaubry committed:
Fixed bug in admin_screen_card.html

pascalaubry committed:
Fixed typo in FR translation

pascalaubry committed:
Fixed the presentation of the roadmap

pascalaubry committed:
Replaced buttons par dropdown menus for screen and family creation

pascalaubry committed:
Fixed typo in FR translation

pascalaubry committed:
Fixed typo in FR translation

pascalaubry committed:
Fixed typo in FR translation

pascalaubry committed:
Removed empty message

pascalaubry committed:
Added instructions for the translators

pascalaubry committed:
Added the date of the MO files to 86-i18n.md
Fixed FR translation

pascalaubry committed:
Added TODO tags for not untranslated strings

Amaras committed:
reorganised pyproject optional-dependencies

Amaras commented on src/data/chessevent_player.py:
```diff
@@ -84,14 +84,12 @@ def __init__(self, chessevent_player_info: dict[str, bool | str | int | dict[int
         except KeyError:
-            logger.error('Champ %s non trouvé pour le·la joueur·euse [%s %s]',
-                         key, self.last_name, self.first_name)
+            logger.error('Field [%s] not found for player [%s %s]', key, self.last_name, self.first_name)
```
This should be translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/data/chessevent_player.py:
```diff
@@ -84,14 +84,12 @@ def __init__(self, chessevent_player_info: dict[str, bool | str | int | dict[int
-                'Valeur du champ %s non valide ([%s]) '
-                'pour le·la joueur·euse [%s %s]',
-                key, chessevent_player_info[key], self.last_name, self.first_name)
+                'Invalid value [%s] for field [%s] for player [%s %s]',
```
This should be translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/data/chessevent_tournament.py:
```diff
@@ -60,12 +60,12 @@ def __init__(
                 self.players.append(chessevent_player)
         except KeyError:
-            logger.error('Champ %s non trouvé dans la réponse de Chess Event', key)
+            logger.error('Field [%s] missing in the ChessEvent response', key)
```
This is not translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/data/chessevent_tournament.py:
```diff
@@ -60,12 +60,12 @@ def __init__(
             logger.error(
-                'Valeur du champ %s non valide ([%s]) '
-                'dans la réponse de Chess Event', key, chessevent_tournament_info[key])
+                'Invalid value [%s] for field [%s] in the ChessEvent response',
```
This needs translation

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/data/family.py:
```diff
@@ -264,36 +273,64 @@ def calculated_parts(self) -> int | None:
     @property
     def numbers_str(self):
         name: str = 'échiquiers' if self.type in [ScreenType.Boards, ScreenType.Input, ] else 'joueur·euses'
```
either `name` is used somewhere and must be translated, or it is not, and should be removed

pascalaubry replied:
Removed.

Amaras commented on src/data/tournament.py:
```diff
@@ -551,7 +549,7 @@ def add_result(self, board: Board, white_result: Result):
             event_database.add_stored_result(self.id, self.current_round, board, white_result)
             event_database.commit()
-        logger.info('Added result: %s %s %d.%d %s %s %d %s %s %s %d',
+        logger.info('Added result: %s %s %d.%d %s %s %d %s %s %s %d.',
```
Should be translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/data/tournament.py:
```diff
@@ -566,7 +564,7 @@ def delete_result(self, board: Board):
             event_database.delete_stored_result(self.id, self.current_round, board.id)
             event_database.commit()
-        logger.info('Removed result: %s %s %d.%d',
+        logger.info('Removed result: %s %s %d.%d.',
```
should be translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/database/access.py:
```diff
@@ -26,23 +26,21 @@ class AccessDatabase:
         needed_driver: str = access_driver()
         if needed_driver not in pyodbc.drivers():
-            logger.error('Les pilotes ODBC installés sont les suivants :')
+            logger.error('ODBC installed drivers are:')
```
```suggestion
            logger.error('Installed ODBC drivers are:')
```
Should also be translated

pascalaubry replied:
No translations for log strings, don't want to recieve logs in German, Dutch...

Amaras commented on src/database/access.py:
```diff
@@ -26,23 +26,21 @@ class AccessDatabase:
-                         'utiliser la commande suivante à l\'installation :')
-            logger.error('accessdatabaseengine_X64.exe /passive')
-            raise PapiWebException('Pilote Microsoft Access introuvable')
+            logger.error('Install the driver (see %s) and restart.', install_url)
+            logger.error('Note: 32/64bits compatibility: accessdatabaseengine_X64.exe /passive')
+            raise PapiWebException('Microsoft Access driver not found.')
```
Missing translation

Amaras reviewed (changes requested):
I stopped before reviewing everything, but a lot of translations are missing, even where log messages were translated.

pascalaubry committed:
Fixed english string

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Improved i18n_update

pascalaubry committed:
Added user Jules MORATA

pascalaubry committed:
Web:
- Added the Players tab
- Added macros admin_players_sort_link, ffe_licence, player_history and gender_icon
- Added a Fide SVG image
I18n:
- Fixed FR translations
- Translated error messages from FR to EN
Requests and sessions:
- Normalized the names of the query parameters of the admin pages
- Normalized the session keys and methods
- Added cache_control to create_static_files_router, still not not caching static files :-(
Data classes:
- Added cached properties du Tournament and Event to count the origins, genders and licences of the players
- Simplified and renamed methods of EventAdminController
- Removed unused string in family.py
- Added classes FederationTuple, LeagueTuple and ClubTuple to ease player sorting
- Made the player ID unique at event-level
- Added fields Player.fide_id, ffe_id, ffe_licence_number, federation, league, club
- Renamed Player.licence_type to ffe_licence
- Renamed PlayerFFELicense to PlayerFFELicence
- Fixed a bug in Tournament.players_by_name_*() (exempt players were not extracted from the database and the first player of the list was ignored)
- Renamed Result.NOT_PAIRED TO Result.NO_RESULT
- Added Tournament methods to move and delete players

pascalaubry committed:
Fixed a bug on the creation of the default screens

pascalaubry committed:
Fixed a string in export.py

pascalaubry committed:
Updated the ChangeLog

pascalaubry committed:
Fixed the width of the columns dropdown

pascalaubry committed:
Improved the display of the players' history

pascalaubry committed:
Let the users choose the columns they display on the players page
Moved fide.svg to images/federations

pascalaubry committed:
Fixed translations

pascalaubry committed:
Fixed player dropdown menu

pascalaubry committed:
Fixed detail in admin_players.html

pascalaubry committed:
Fixed forgotten translation

pascalaubry committed:
Added the possibility to update the players' identity

pascalaubry committed:
Fixed ZeroDivisionError when specifying the time control of a tournament with no penalty step

pascalaubry committed:
Added the possibility to update the players' ratings and FIDE title
Use upper/capwords for the players' last and first names
Enlarged all the modals to 90%
Replaced 'Fide' by 'FIDE' everywhere

pascalaubry committed:
Added the possibility to update the players' federation and FIDE ID
Added the GRL flag
Added lib dynamic-select to add images to select options
Listed all the federations in PapiWebConfig

pascalaubry committed:
Fixed FR translation

pascalaubry committed:
Fixed macro ffe_licence()

pascalaubry committed:
Added the possibility to edit the other players' properties (FFE, contact and registration)

pascalaubry committed:
Improved methods print_interactive_*() with an end parameter

pascalaubry committed:
Fixed a bug when moving players from one tournament to another one

pascalaubry committed:
Fixed a bug with Tournament.time_control_handicap_penalty_value

Amaras commented on src/web/static/lib/dynamic-select/dynamic-select.js:
```diff
@@ -0,0 +1,183 @@
+            </div>
+        `;
+        let element = document.createElement('div');
+        element.innerHTML = template;
```
I don't have enough information to dismiss the alert or commit the suggestion.
Where does the `'name'` attribute comes from?

If it is user-controlled, this may be vulnerable to XSS attacks.

pascalaubry replied:
I think you can dismiss, name is hard-coded in admin_player_modal.html: {% with id='federation', name='federation' %}.

Amaras replied:
I don't know about that.
There are several places where I saw dynamic-select stuff

Amaras commented on src/common/papi_web_config.py:
```diff
@@ -38,295 +31,697 @@ class PapiWebConfig(metaclass=Singleton):
+        'ZIM': 'Zimbabwe',
+    }
+
+    """ The federation names. """
```
Federation names for FFE regional leagues?

Amaras commented on src/data/family.py:
```diff
@@ -263,37 +268,68 @@ def calculated_parts(self) -> int | None:
-            case _:
-                raise ValueError(
-                    f'first={self.first}, last={self.last}, parts={self.parts}, number={self.number}')
+        if self.type in [ScreenType.Boards, ScreenType.Input]:
```
You don't need to use a list when you can use a tuple:
```suggestion
        if self.type in (ScreenType.Boards, ScreenType.Input):
```

Amaras commented on src/data/pairing.py:
```diff
@@ -11,10 +12,54 @@ class Pairing:
+        return (self.result == Result.NO_RESULT) and (self.opponent_id is not None)
+
+    @property
+    def exempt(self) -> bool:
```
```suggestion
    def pairing_allocated_bye(self) -> bool:
```

Amaras commented on src/data/pairing.py:
```diff
@@ -11,10 +12,54 @@ class Pairing:
 
+    @property
+    def not_paired(self) -> bool:
+        return (self.result == Result.NO_RESULT) and (self.opponent_id is None)
```
`self.opponent_id` or `self.opponent_papi_id`?

Amaras commented on src/data/pairing.py:
```diff
@@ -11,10 +12,54 @@ class Pairing:
+
+    @property
+    def playing(self) -> bool:
+        return (self.result == Result.NO_RESULT) and (self.opponent_id is not None)
```
`self.opponent_id` or `self.opponent_papi_id`?

Amaras commented on src/data/player.py:
```diff
@@ -176,7 +340,7 @@ def __eq__(self, other):
-        if self.id == 1:
-            return f'{self.__class__.__name__}(EXEMPT)'
+        if self.ref_id == 1:
+            return f'{self.__class__.__name__}(#{self.id} EXEMPT)'
```
I don't think we should still use `EXEMPT`in the representation, we should probably use `PAB`instead:
```suggestion
            return f'{self.__class__.__name__}(#{self.id} PAB)'
```

Amaras commented on src/data/tournament.py:
```diff
@@ -350,14 +398,12 @@ def _calculate_current_round(self):
-                    if color in ['W', 'B', ]:
+                if player.ref_id != 1:
+                    pairing: Pairing = player.pairings[round_]
+                    if pairing.color in ['W', 'B', ]:
```
```suggestion
                    if pairing.color in ('W', 'B', ):
```

Amaras commented on src/data/tournament.py:
```diff
@@ -350,14 +398,12 @@ def _calculate_current_round(self):
-                    # NOTE(Amaras) Why is it called RESULT_NOT_PAIRED if it also
-                    # represents a missing result?
-                    if result == Result.NOT_PAIRED and opponent_id is not None:
+                    if pairing.result == Result.NO_RESULT and pairing.opponent_id is not None:
```
Now that we can use properties in `data.pairing.Pairing`, why not use them?

Amaras commented on src/data/tournament.py:
```diff
@@ -579,17 +625,17 @@ def write_chessevent_info_to_database(
+                player_papi_id: int = 1
                 for chessevent_player in chessevent_tournament.players:
-                    player_id += 1
+                    player_papi_id += 1
                     event_database.add_player_stored_skipped_rounds(
-                        self.id, player_id, chessevent_player.skipped_rounds)
+                        self.id, player_papi_id, chessevent_player.skipped_rounds)
                     papi_database.add_chessevent_player(
-                        player_id, chessevent_player, chessevent_tournament.check_in_started)
+                        player_papi_id, chessevent_player, chessevent_tournament.check_in_started)
```
This should probably use `enumerate` instead:
`for player_papi_id, chessevent_player in enumerate(chessevent_tournament, start=2):`

Amaras commented on src/data/util.py:
```diff
@@ -425,6 +425,10 @@ class PlayerGender(IntEnum):
 
+    @classmethod
+    def values(cls) -> list[int]:
+        return [item.value for item in cls]
```
```suggestion
    def values(cls) -> tuple[int]:
        return tuple(item.value for item in cls)
```

Amaras commented on src/database/access.py:
```diff
@@ -26,23 +26,21 @@ class AccessDatabase:
             for driver in odbc_drivers():
                 logger.error(' - %s', driver)
-            logger.error('Pilote nécessaire : %s', needed_driver)
+            logger.error('Driver needed : %s', needed_driver)
```
```suggestion
            logger.error('Needed driver: %s', needed_driver)
```

Amaras commented on src/database/papi.py:
```diff
@@ -264,68 +394,69 @@ def _check_in_player(self, player_id: int, tournament_skipped_rounds_dict: dict[
         actions: str = ', '.join([f'`{key}` = ?' for key in data.keys()])
         query: str = f'UPDATE `joueur` SET {actions} WHERE Ref = ?'
-        params = tuple(list(data.values()) + [player_id, ])
+        params = tuple(list(data.values()) + [player_papi_id, ])
```
```suggestion
        params = tuple(data.values()) + (player_papi_id, ))
```

Amaras commented on src/web/controllers/admin/player_admin_controller.py:
```diff
@@ -0,0 +1,504 @@
+        else:
+            last_name = last_name.upper()
+        field: str = 'first_name'
+        first_name: str = WebContext.form_data_to_str(data, field)
+        if not first_name:
+            errors[field] = _('Please enter the first name.')
+        else:
+            first_name = string.capwords(first_name)
```
Absolutely not: not all people have first names!
If the first name does not exist, it should not be an error, and should simply give an empty first name.
At most, it should be a warning to make sure it's wanted, and even then, it's a bit much.

Amaras commented on src/web/controllers/admin/player_admin_controller.py:
```diff
@@ -0,0 +1,504 @@
+        event_loader.clear_cache(event_uniq_id)
+        return self._admin_event_players_render(request, event_uniq_id=event_uniq_id)
+
+    """
+    @post(
+        path='/admin/tournament-create/{event_uniq_id:str}',
+        name='admin-tournament-create'
+    )
+    async def htmx_admin_tournament_create(
+            self, request: HTMXRequest,
+            data: Annotated[dict[str, str], Body(media_type=RequestEncodingType.URL_ENCODED), ],
+            event_uniq_id: str,
+    ) -> Template | ClientRedirect:
+        return self._admin_tournament_update(
+            request, event_uniq_id=event_uniq_id, action='create', tournament_id=None, data=data)
+    """
```
Well, we'll see what happens from now

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -154,7 +161,64 @@ def _form_data_to_rgb(self, field: str, empty_value: RGB | None = None) -> str |
+            data[field] = data[field].strip().lower()
+        if not data[field]:
+            return None
+        if re.match(r'^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}$', data[field]):
```
There is no regex able to validate emails, even if this one is more lax than most ones found online.

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -154,7 +161,64 @@ def _form_data_to_rgb(self, field: str, empty_value: RGB | None = None) -> str |
+            return None
+        if re.match(r'^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}$', data[field]):
+            return data[field]
+        raise ValueError
```
Not enough context there

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -154,7 +161,64 @@ def _form_data_to_rgb(self, field: str, empty_value: RGB | None = None) -> str |
+                phonenumbers.parse(data[field], get_locale().upper())
+                return data[field]
+            except NumberParseException:
+                raise ValueError
```
Not enough context there

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -154,7 +161,64 @@ def _form_data_to_rgb(self, field: str, empty_value: RGB | None = None) -> str |
+            return None
+        if re.match(r'^[A-Za-z][0-9]{5}$', data[field]):
+            return data[field]
+        raise ValueError
```
Not enough context there

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -163,15 +227,27 @@ def value_to_form_data(value: str | int | bool | Path | None) -> str | None:
+            return f'{value:.2f}'
         if isinstance(value, Path):
             return str(value)
         raise ValueError
```
not enough context there

Amaras commented on src/web/controllers/index_controller.py:
```diff
@@ -163,15 +227,27 @@ def value_to_form_data(value: str | int | bool | Path | None) -> str | None:
+            return datetime.strftime(datetime.fromtimestamp(value), '%Y-%m-%dT%H:%M')
+        if isinstance(value, datetime):
+            return datetime.strftime(value, '%Y-%m-%dT%H:%M')
+        raise ValueError
```
not enough context there

Amaras commented on src/web/templates/user_boards_screen_set.html:
```diff
@@ -14,9 +20,9 @@ <h2 class="set-title text-center">{{ screen_set.name_for_boards }} (ronde {{ tou
                                         {% if tournament.print_real_points %}<th scope="col" class="points">&nbsp;</th>{% endif %}
-                                        <th scope="col" class="player w-50" {% if screen.type == 'input' and tournament.record_illegal_moves %}colspan="2">{% endif %}Blancs{% if tournament.handicap %} (temps){% endif %}</th>
+                                        <th scope="col" class="player w-50" {% if screen.type == 'input' and tournament.record_illegal_moves %}colspan="2">{% endif %}{{ _('White') }}{% if tournament.handicap %} ({{ _('time control') }}){% endif %}</th>
                                         <th scope="col" class="score">Score</th>
-                                        <th scope="col" class="player w-50" {% if screen.type == 'input' and tournament.record_illegal_moves %}colspan="2">{% endif %}Noirs{% if tournament.handicap %} (temps){% endif %}</th>
+                                        <th scope="col" class="player w-50" {% if screen.type == 'input' and tournament.record_illegal_moves %}colspan="2">{% endif %}{{ _('Black') }}{% if tournament.handicap %} ({{ _('time control') }}){% endif %}</th>
```
I would prefer if both white and black were visible on all input or boards screen set

Amaras commented on src/web/templates/user_rotator_card.html:
```diff
@@ -15,31 +15,37 @@
     <div class="mb-2">
         {% if not rotator.families and not rotator.screens %}
-            Aucun écran ni famille à afficher.
+            {{ _('No screen neither family to rotate.') }}
```
```suggestion
            {{ _('No screen nor family to rotate.') }}
```
But we might need to correct a bunch of stuff to have less weird grammar

Amaras commented on src/web/templates/user_timer.html:
```diff
@@ -1,6 +1,6 @@
 <div class="timer-wrapper">
     <div id="timer" class="row timer w-100 m-0 fw-bold">
-        <h2 id="timer-text" class="col-10 timer-text text-start">Texte</h2>
+        <h2 id="timer-text" class="col-10 timer-text text-start"></h2>
```
Are you sure you want an empty h2?

Amaras reviewed (changes requested):
Here goes, I read all files.
I have not tried to run the code yet.

Amaras committed:
Added phonenumbers dependency

Amaras commented:
Now for the problems found while running the code (I would like to have them solved before merging to dev):

First:
In the event admin page, the "Timers" and "ChessEvent" tabs do not work, see below for `chessevents`
```File "C:\Users\Sammy\PycharmProjects\papi-web\src\web\controllers\admin\event_admin_controller.py", line 540, in htmx_admin_event_tab
    raise ValueError(f'admin_event_tab={admin_event_tab}')
ValueError: admin_event_tab=chessevents```

Second:
I can't create a tournament any more, apparently because I set up that tournaments have a maximum of 3 illegal moves:
```File "C:\Users\Sammy\PycharmProjects\papi-web\src\web\controllers\admin\tournament_admin_controller.py", line 269, in _admin_event_tournaments_render
    'record_illegal_moves_options': cls._get_record_illegal_moves_options(
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sammy\PycharmProjects\papi-web\src\web\controllers\admin\index_admin_controller.py", line 82, in _get_record_illegal_moves_options
    options[''] = _('By default - {option}').format(option=options[str(default)])
                                                           ~~~~~~~^^^^^^^^^^^^^^
KeyError: '3'```
Related to that, a weird bug that changes the number of illegal moves back to 0 (default) when I edit it back, even after changing it.
The tournament creation bug only happens when the event has a non-zero number of illegal moves set.

Third (could be done later, as long as it's not released in that state in a stable release):
In the image below is shows the fact that the filtering of clubs is not usable: only the background scrolls, while the dropdown doesn't, even when trying to scroll on the dropdown.
![2025-01-16 02 55 17 127 0 0 1 47be48119e88](https://github.com/user-attachments/assets/99210d05-a25e-4221-86ce-9cb3d4504b3e)

I'll stop right there for now, the first two are quite fundamental.

pascalaubry committed:
Added the possibility to check-in players from the admin pages
Added Tournament.check_in_open to control check-in at Tournament-level
Moved the check-in column next to the name column on the Players tab
Set the players forfeit for already paired rounds when moving to another tournament
Fixed bug on the ChessEvent and Timers tab names
Version 2.4.20

pascalaubry committed:
Merge remote-tracking branch 'origin/i18n' into i18n

pascalaubry committed:
Added missing translations

pascalaubry committed:
Fixed the Timers tab name

pascalaubry committed:
Fixed the check-in status tooltip

pascalaubry committed:
Restored event cloning

pascalaubry committed:
Apply suggestions from code review

Sammy's review

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Merge branch 'dev' into i18n

pascalaubry committed:
Fixed White and Black screens on boards screens
Fixed empty h2 tag in user_timer.html
Fixed ngettext bug in admin_tournament_card.html
Fix tournament creation bug when 1 to 3 illegal moves are recorded
Added Result.ZERO_POINT_BYE

pascalaubry committed:
Fixed White and Black screens on boards screens
Fixed empty h2 tag in user_timer.html
Fixed ngettext bug in admin_tournament_card.html
Fix tournament creation bug when 1 to 3 illegal moves are recorded
Added Result.ZERO_POINT_BYE

pascalaubry committed:
Fixed the pairing colors

pascalaubry committed:
Do not display the tournament column on the Players tab when there is a single tournament in the event

Amaras reviewed (approved):
LGTM
